### ISSUE TITLE
Cannot force install python anymore again (master)

### ISSUE THREAD

**Issue by ilanschnell** [conda/conda#2554]
On OSX, I get:
```
$ conda install -f python
...
Traceback (most recent call last):
  File "/Users/ilan/python/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main())
  File "/Users/ilan/conda/conda/cli/main.py", line 120, in main
    args_func(args, p)
  File "/Users/ilan/conda/conda/cli/main.py", line 127, in args_func
    args.func(args, p)
  File "/Users/ilan/conda/conda/cli/main_install.py", line 62, in execute
    install.install(args, parser, 'install')
  File "/Users/ilan/conda/conda/cli/install.py", line 309, in install
    update_deps=args.update_deps)
  File "/Users/ilan/conda/conda/plan.py", line 450, in install_actions
    force=force, always_copy=always_copy)
  File "/Users/ilan/conda/conda/plan.py", line 311, in ensure_linked_actions
    elif install.try_hard_link(fetched_dir, prefix, dist):
  File "/Users/ilan/conda/conda/install.py", line 579, in try_hard_link
    assert isfile(src), src
AssertionError: /Users/ilan/python/pkgs/pyt/info/index.json
```

**Comment by ilanschnell** [comment]
Upon further inspection, I see that the function `_dist2filename()`, returns `pyt`, which obviously does not correspond to an extracted package.

CC @kalefranz @mcg1969 @csoja

**Comment by mcg1969** [comment]
Grr, thanks for finding. Perhaps a merge conflict

**Comment by mcg1969** [comment]
No, that's a legit regression from the `urlmeta` PR. Very simple fix, and I will add some tests

**Comment by mcg1969** [comment]
Back to sanity:
```
$ conda install -f python
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata .......
Solving package specifications .............

Package plan for installation in environment /Users/mgrant/miniconda2:

The following NEW packages will be INSTALLED:

    python: 2.7.11-0
```
Fixed via #2555, which I have taken the liberty to merge.

**Comment by ilanschnell** [comment]
@mcg1969 Thanks for the fix!  I can confirm that it works now.
